# Class 11 : Computer Science Practical File

**Name :** Jai Relan

**Class & Section :** 11 - B

**Date of Submission :** 25th February 2022

**Teachers Name :** Ms Sunita Singh

**Format :** All questions are solved sequentially. The markdown box contains the question followed by the code box containing the code. In questions which require an input, I've included the input as a comment within the box containing the code or I've tried to print it out along with the output (since I'm worked on Jupyter Notebook and the input is not displayed natively).

**Export Instructions :** The code is written within PyCharm's Jupyter Notebook (exported as a .ipynb) - in order to compile this as a PDF, you open the notebook's Jupyter instance on Localhost (http://localhost:8888) and click on the "Export to LaTeX" button on the File > Download As menu. Now, go to Overleaf (https://overleaf.com), sign in, and create a blank file in which you can paste the content which you downloaded in the LaTex file.

**Index :**
- [x] Q1
- [x] Q2
- [x] Q3
- [x] Q4
- [ ] Q5
- [ ] Q6
- [ ] Q7
- [ ] Q8
- [ ] Q9
- [ ] Q10
- [ ] Q11
- [ ] Q12
- [ ] Q13
- [ ] Q14
- [ ] Q15
- [ ] Q16
- [ ] Q17
- [ ] Q18
- [ ] Q19
- [ ] Q20
- [ ] Q21
- [ ] Q22
- [ ] Q23
- [ ] Q24
- [ ] Q25
- [ ] Q26
- [ ] Q27
- [ ] Q28
- [ ] Q29
- [ ] Q30
- [ ] Q31
- [ ] Q32
- [ ] Q33
- [ ] Q34
- [ ] Q35
- [ ] Q36
- [ ] Q37
- [ ] Q38
- [ ] Q39
- [ ] Q40
- [ ] Q41
- [ ] Q42
- [ ] Q43
- [ ] Q44
- [ ] Q45
- [ ] Q46
- [ ] Q47
- [ ] Q48
- [ ] Q49
- [ ] Q50
- [ ] Q51
- [ ] Q52
- [ ] Q53
- [ ] Q54
- [ ] Q55
- [ ] Q56
- [ ] Q57
- [ ] Q58
- [ ] Q59
- [ ] Q60
- [ ] Q61
- [ ] Q62
- [ ] Q63
- [ ] Q64
- [ ] Q65
- [ ] Q66
- [ ] Q67
- [ ] Q68
- [ ] Q69
- [ ] Q70
- [ ] Q71
- [ ] Q72
- [ ] Q73
- [ ] Q74
- [ ] Q75
- [ ] Q76
- [ ] Q77
- [ ] Q78
- [ ] Q79
- [ ] Q80

Print the even numbers between 1 and 100.

In [13]:
# Q1
for i in range(0 ,101, 2):
    print(i)

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


Print the odd numbers between 30 and 400.

In [9]:
# Q2
for i in range(31, 401, 2):
    print(i)

31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99
101
103
105
107
109
111
113
115
117
119
121
123
125
127
129
131
133
135
137
139
141
143
145
147
149
151
153
155
157
159
161
163
165
167
169
171
173
175
177
179
181
183
185
187
189
191
193
195
197
199
201
203
205
207
209
211
213
215
217
219
221
223
225
227
229
231
233
235
237
239
241
243
245
247
249
251
253
255
257
259
261
263
265
267
269
271
273
275
277
279
281
283
285
287
289
291
293
295
297
299
301
303
305
307
309
311
313
315
317
319
321
323
325
327
329
331
333
335
337
339
341
343
345
347
349
351
353
355
357
359
361
363
365
367
369
371
373
375
377
379
381
383
385
387
389
391
393
395
397
399


To input a number and chek if it's a multiple of 3.

In [1]:
# Q3
input_number = int(input("Enter a number: "))
print("Given input", input_number, "is a multiple of 3: " + str(input_number % 3 == 0))

# Input given by user = 6

Given input 12 is a multiple of 3: True


To print the square of numbers from 5 to 50.

In [11]:
# Q4
for i in range(5, 51):
    print(i**2)

25
36
49
64
81
100
121
144
169
196
225
256
289
324
361
400
441
484
529
576
625
676
729
784
841
900
961
1024
1089
1156
1225
1296
1369
1444
1521
1600
1681
1764
1849
1936
2025
2116
2209
2304
2401
2500


Write a program that allows an integer to be entered by the user. The program should then count the number of times the number has to be doubled before it reaches one million.

In [ ]:
# Q32
input_integer = int(input("Enter an integer: "))
count = 0
while input_integer < 1000000:
    input_integer *= 2
    count += 1
print(count)

To calculate the square roots of all integers between 1 and 500 and print only
those roots which are integers(whole numbers).


In [ ]:
# Q35
for i in range(1, 501):
    if i ** 0.5 % 1 == 0:
        print(i)

The expression rand( )%100 always generates a number between 0 and 99. Generate 100 such numbers and print out the number of numbers in each decile (0 to 9, 10 to 19, 20 to 29 etc.)

In [3]:
# Q38
import random
random_numbers = {
    "0-9": 0,
    "10-19": 0,
    "20-29": 0,
    "30-39": 0,
    "40-49": 0,
    "50-59": 0,
    "60-69": 0,
    "70-79": 0,
    "80-89": 0,
    "90-99": 0
}

for i in range(101):
    rand = int(random.randint(0, 99) / 10) * 10
    random_numbers[str(rand) + "-" + str(rand + 9)] += 1

print(random_numbers)

{'0-9': 8, '10-19': 9, '20-29': 11, '30-39': 8, '40-49': 7, '50-59': 13, '60-69': 10, '70-79': 8, '80-89': 11, '90-99': 16}


Write a program to generate 1000 random numbers between 5 and 20. Prepare a
frequency table for the random numbers lying between (i) 6 – 10 (ii) 11 – 15 and
(iii) 16 – 20. also print the random numbers occurring in each interval separately.

In [2]:
# Q41
import random
random_numbers = {
    "6-10": 0,
    "11-15": 0,
    "16-20": 0
}
for i in range(1, 1001):
    rand = random.uniform(5, 20)
    if rand >= 6 and rand <= 10:
        random_numbers['6-10'] += 1
    elif rand >= 11 and rand <= 15:
        random_numbers['11-15'] = random_numbers.get('11-15') + 1
    elif rand >= 16 and rand <= 20:
        random_numbers['16-20'] = random_numbers.get('16-20') + 1

print("Frequency table >")
print("6-10:", random_numbers.get('6-10'))
print("11-15:", random_numbers.get('11-15'))
print("16-20:", random_numbers.get('16-20'))

Frequency table >
{'6-10': 237, '11-15': 276, '16-20': 276}
6-10: 237
11-15: 276
16-20: 276


The number 151 is a prime palindrome because it is both a prime number and a
palindrome. Write a program that finds all prime palindromes between two
numbers a and b. You may assume that a and b are between 1 and 5000.

In [5]:
# Q44
def isprime(number):
    if number == 1:
        return False
    for i in range(2, number):
        if number % i == 0:
            return False
    return True

for number in range(int(input("Please input an integer 'a' > ")), int(input("Please inbput an integer 'b' > "))):
    if isprime(number) and str(number) == str(number)[::-1]:
        print(number)

101
131
151
181
191
313
353
373
383
727
757
787
797
919
929


Write a program that prompts the user to enter 5 exam marks (design your
program such that it would be easy to change if we wanted it to work for more
than 5 exams). The program then displays the average mark, the highest mark,
and number of marks entered less than the average mark. For example, when
executing the program and inputting 20, 30, 10, 40, and 50 the running session
will look as follows:

Enter a mark: 20

Enter a mark: 30

Enter a mark: 10

Enter a mark: 40

Enter a mark: 50

Average mark is: 30.0

Highest mark is: 50

Number of marks less than the average: 2

In [6]:
# Q47
marks = []
for exam in range(1, 6): #replace 6 with the number of exams you want to input
    marks.append(int(input("Enter a mark: ")))
print("Average mark is:", sum(marks)/len(marks))
print("Highest mark is:", max(marks))
print("Number of marks less than the average:", len([mark for mark in marks if mark < sum(marks)/len(marks)]))

Average mark is: 25.0
Highest mark is: 40
Number of marks less than the average: 2


Read two integers, representing a rate of pay (pence per hour) and a number of
hours. Print out the total pay, with hours up to 40 being paid at basic rate, from 40
to 60 at rate-and-a-half, above 60 at double-rate. Print the pay as pounds to two
decimal places.

Terminate the loop when a zero rate is encountered. At the end of the
loop, print out the total pay.
The code for computing the pay from the rate and hours is to be written
as a function.

The recommended output format is something like:

Pay at 200 pence/hr for 38 hours is 76.00 pounds

Pay at 220 pence/hr for 48 hours is 114.40 pounds

Pay at 240 pence/hr for 68 hours is 206.40 pounds

Pay at 260 pence/hr for 48 hours is 135.20 pounds

Pay at 280 pence/hr for 68 hours is 240.80 pounds

Pay at 300 pence/hr for 48 hours is 156.00 pounds

Total pay is 928.80 pounds

In [7]:
# Q50
rate, hours = int(input("Enter the rate: ")), int(input("Enter the hours: "))
def pay(rate, hours):
    if hours <= 40:
        return rate * hours
    elif hours > 40 and hours <= 60:
        return rate * 40 + (rate * 1.5 * (hours - 40))
    elif hours > 60:
        return rate * 40 + (rate * 1.5 * 40) + (rate * 2 * (hours - 60))
print("Pay at", rate, "pence/hr for", hours, "hours is", pay(rate, hours), "pounds")

Pay at 200 pence/hr for 48 hours is 10400.0 pounds


Write a program which inputs an integer number representing the age of a person
(using the class SimpleInput), and prints the number out followed by the word
"Minor" if the number (age) is less than 18, the word "Pensioner" if the number is
greater than 65, and the word "Tax-payer" otherwise.

In [ ]:
# Q51
import simpleinput
simpleinput.Input()

Write an algorithm to produce an n times multiplication table ( n less than or
equal to 10).

In [9]:
# Q54
n = int(input("Enter a number: "))
for i in range(1, n+1):
    for j in range(1, n+1):
        print(i * j, end = " ")
    print()

1 2 3 4 
2 4 6 8 
3 6 9 12 
4 8 12 16 


Write a program that inputs numbers into an array and then inputs a constant. The
program should count the number of elements that are less than the constant,
equal to it and greater than it.

In [13]:
# Q57
arr = list(map(int, input("Enter a list of numbers: ").split()))
constant = int(input("Enter a constant: "))
print("Number of elements less than the constant:", len([x for x in arr if x < constant]))
print("Number of elements equal to the constant:", len([x for x in arr if x == constant]))
print("Number of elements greater than the constant:", len([x for x in arr if x > constant]))

Number of elements less than the constant: 0
Number of elements equal to the constant: 1
Number of elements greater than the constant: 3


Write a program to input the examination marks for a class of 10 students,
studying 6 subjects into a 2-D array. Calculate and print the average marks
obtained by the class in each subject.

In [ ]:
# Q60
marks = [
    [int(input("Marks Person #1 Subject #1: ")),
     int(input("Marks Person #2 Subject #1: ")),
     int(input("Marks Person #3 Subject #1: ")),
     int(input("Marks Person #4 Subject #1: ")),
     int(input("Marks Person #5 Subject #1: ")),
     int(input("Marks Person #6 Subject #1: ")),
     int(input("Marks Person #7 Subject #1: ")),
     int(input("Marks Person #8 Subject #1: ")),
     int(input("Marks Person #9 Subject #1: ")),
     int(input("Marks Person #10 Subject #1: "))],
    [int(input("Marks Person #1 Subject #2: ")),
     int(input("Marks Person #2 Subject #2: ")),
     int(input("Marks Person #3 Subject #2: ")),
     int(input("Marks Person #4 Subject #2: ")),
     int(input("Marks Person #5 Subject #2: ")),
     int(input("Marks Person #6 Subject #2: ")),
     int(input("Marks Person #7 Subject #2: ")),
     int(input("Marks Person #8 Subject #2: ")),
     int(input("Marks Person #9 Subject #2: ")),
     int(input("Marks Person #10 Subject #2: "))],
    [int(input("Marks Person #1 Subject #3: ")),
     int(input("Marks Person #2 Subject #3: ")),
     int(input("Marks Person #3 Subject #3: ")),
     int(input("Marks Person #4 Subject #3: ")),
     int(input("Marks Person #5 Subject #3: ")),
     int(input("Marks Person #6 Subject #3: ")),
     int(input("Marks Person #7 Subject #3: ")),
     int(input("Marks Person #8 Subject #3: ")),
     int(input("Marks Person #9 Subject #3: ")),
     int(input("Marks Person #10 Subject #3: "))],
    [int(input("Marks Person #1 Subject #4: ")),
     int(input("Marks Person #2 Subject #4: ")),
     int(input("Marks Person #3 Subject #4: ")),
     int(input("Marks Person #4 Subject #4: ")),
     int(input("Marks Person #5 Subject #4: ")),
     int(input("Marks Person #6 Subject #4: ")),
     int(input("Marks Person #7 Subject #4: ")),
     int(input("Marks Person #8 Subject #4: ")),
     int(input("Marks Person #9 Subject #4: ")),
     int(input("Marks Person #10 Subject #4: "))],
    [int(input("Marks Person #1 Subject #5: ")),
     int(input("Marks Person #2 Subject #5: ")),
     int(input("Marks Person #3 Subject #5: ")),
     int(input("Marks Person #4 Subject #5: ")),
     int(input("Marks Person #5 Subject #5: ")),
     int(input("Marks Person #6 Subject #5: ")),
     int(input("Marks Person #7 Subject #5: ")),
     int(input("Marks Person #8 Subject #5: ")),
     int(input("Marks Person #9 Subject #5: ")),
     int(input("Marks Person #10 Subject #5: "))],
    [int(input("Marks Person #1 Subject #6: ")),
     int(input("Marks Person #2 Subject #6: ")),
     int(input("Marks Person #3 Subject #6: ")),
     int(input("Marks Person #4 Subject #6: ")),
     int(input("Marks Person #5 Subject #6: ")),
     int(input("Marks Person #6 Subject #6: ")),
     int(input("Marks Person #7 Subject #6: ")),
     int(input("Marks Person #8 Subject #6: ")),
     int(input("Marks Person #9 Subject #6: ")),
     int(input("Marks Person #10 Subject #6: "))]
    ]

print("Average marks in subject #1:", sum(marks[0]) / len(marks[0]))
print("Average marks in subject #2:", sum(marks[1]) / len(marks[1]))
print("Average marks in subject #3:", sum(marks[2]) / len(marks[2]))
print("Average marks in subject #4:", sum(marks[3]) / len(marks[3]))
print("Average marks in subject #5:", sum(marks[4]) / len(marks[4]))
print("Average marks in subject #6:", sum(marks[5]) / len(marks[5]))

Write a program to input a word and check whether it is palindrome or not.

In [ ]:
# Q63
string = input("Enter a word: ")
print(string == string[::-1])

Write a program to input a sentence and delete from it all occurrences of the word
“the” from it.